flink-runtime/src/main/java/org/apache/flink/runtime/state/memory/ByteStreamStateHandle.java

```diff
@@ -20,62 +20,49 @@
 
 import org.apache.flink.core.fs.FSDataInputStream;
 import org.apache.flink.runtime.state.StreamStateHandle;
-import org.apache.flink.util.InstantiationUtil;
 import org.apache.flink.util.Preconditions;
 
 import java.io.IOException;
-import java.io.Serializable;
-import java.util.Arrays;
 
 /**
  * A state handle that contains stream state in a byte array.
  */
 public class ByteStreamStateHandle implements StreamStateHandle {
 
-	private static final long serialVersionUID = -5280226231200217594L;
+	private static final long serialVersionUID = -5280226231202517594L;
 
 	/**
-	 * the state data
+	 * The state data.
 	 */
 	protected final byte[] data;
 
 	/**
+	 * A unique name of by which this state handle is identified and compared. Like a filename, all
+	 * {@link ByteStreamStateHandle} with the exact same name must also have the exact same content in data.
+	 */
+	protected final String handleName;
```

Comment:
Is this meant to be an optimization for `equals/hashCode`? If yes, has it been an issue before?